File: src/main/java/com/github/theword/queqiao/tool/event/BasePlayer.java
package com.github.theword.queqiao.tool.event;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class BasePlayer {
    private String nickname;
    private UUID uuid;

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof BasePlayer)) return false;
        if (this == o) return true;
        if (uuid != null && uuid.equals(((BasePlayer) o).uuid)) return true;
        return nickname != null && nickname.equals(((BasePlayer) o).nickname);
    }

    @Override
    public int hashCode() {
        return nickname.hashCode();
    }
}

File: src/main/java/com/github/theword/queqiao/tool/event/base/BasePlayer.kt
package com.github.theword.queqiao.tool.event.base

import com.google.gson.Gson
import java.util.UUID

data class BasePlayer(private val nickname: String, private val uuid: UUID? = null) {

    override fun equals(other: Any?): Boolean {
        if (other !is BasePlayer) return false
        if (this === other) return true
        if (uuid != null && uuid == other.uuid) return true
        return nickname == other.nickname
    }

    override fun hashCode(): Int {
        return nickname.hashCode()
    }

    val json: String
        get() = Gson().toJson(this)
}
